content/Constructor/Constructor.q.md

```diff
@@ -0,0 +1,1703 @@
+
+ - It accesses the value of `$name` in the `$name` property of the `Person` class.
+
+ - It retrieves the value of `$name` in the `$name` property of the `Person` class.
+
+ - It evaluates the value of `$name` in the `$name` property of the `Person` class.
+
+ - It removes the value of `$name` from the `$name` property of the `Person` class.
+
+
+/// type=MS, answer=[1,5]
+
+On lines 6, 7, 8, and 9, what does the `__construct()` method do?
+
+ - It sets the value of the `Person` class property `$name`.
+
+ - It returns the value of the `Person` class property `$name`.
+
+ - It removes the value of the `Person` class property `$name`.
+
+ - It accesses the value of the `Person` class property `$name`.
+
+ - It defines the constructor of the `Person` class with the parameter `$name`.
+
+
+/// type=MS, answer=[1,5]
+
+Which statements correctly describe `$pObject = new Person("John");` on line 11?
+
+ - It initializes the `$name` property of the `$pObject` object.
+
+ - It removes the argument `John` from the `$pObject` object.
+
+ - It returns the `$pObject` object of the `Person` class with the parameter `John`.
+
+ - It declares the `$pObject` object of the `Person` class with the parameter `John`.
+
+ - It creates the `$pObject` object of the `Person` class passing the argument `John`.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+class Person
+{
+    public $name;
+    private $age;
+	
+	public function __construct($name, $age)
+	{
+		$this->name = $name;
+		if($this->checkAge($age)) {
+			$this->age = $age;
+		}
+    }
+    
+    public function setAge($newAge)
+    {
+        if ($this->checkAge($newAge)) {
+            $this->age = $newAge;
+        }
+    }
+
+    public function getAge()
+    {
+        return $this->age;
+    }
+
+    private function checkAge($age)
+    {
+        if ($age > 0) {
+            return true;
+        }
+        return false;
+    }
+
+    public function display()
+    {
+        echo $this->name . " is " . $this->getAge() . " years old.";
+    }
+}
+$pObject = new Person("James", 15);
+$pObject->display();
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+ - It prints `15`.
+
+ - It prints `James`.
+
+ - No output is displayed.
+
+ - It prints `James is years old.`.
+
+ - It prints `James is 15 years old.`.
+
+
+/// type=MS, answer=[3,4]
+
+Which of the following are visibility keywords?
+
+ - `new`
+
+ - `class`
+
+ - `public`
+
+ - `private`
+
+ - `function`
+
+
+/// type=SS, answer=[4]
+
+Which method is accessible only within the `Person` class?
+
+ - `setAge()`
+
+ - `getAge()`
+
+ - `display()`
+
+ - `checkAge()`
+
+ - `__construct()`
+
+
+/// type=SS, answer=[1]
+
+Which property is accessible only within the `Person` class?
+
+ - `$age`
+
+ - `$name`
+
+ - `Person`
+
+ - `$newAge`
+
+ - `$pObject`
+
+
+/// type=SS, answer=[5]
+
+Which of the following is a method definition of the `Person` class constructor?
```

Comment:
Sorry, but I opt to retain this question since a class constructor is also a method so generally my question statement here is correct. Please verify. Thanks.